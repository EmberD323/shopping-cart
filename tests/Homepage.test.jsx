
import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import HomePage from "../src/components/Homepage";

describe("Homepage component", () => {
  it("renders brand name and an image", () => {
    // since screen does not have the container property, we'll destructure render to obtain a container for this test
    const { container } = render(<HomePage />);
    expect(container).toMatchSnapshot();
  });

//   it("renders radical rhinos after button click", async () => {
//     const user = userEvent.setup();

//     render(<App />);
//     const button = screen.getByRole("button", { name: "Click Me" });

//     await user.click(button);

//     expect(screen.getByRole("heading").textContent).toMatch(/radical rhinos/i);
//   });
});
