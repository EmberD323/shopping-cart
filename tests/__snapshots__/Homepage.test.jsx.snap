// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > renders brand name and an image 1`] = `
<div>
  <img
    alt="Dummy image"
    src="https://picsum.photos/300/300"
  />
  <p>
    This is the Home Page
  </p>
</div>
`;

exports[`App component > renders this is homepage 1`] = `
<div>
  <img
    alt="Dummy image"
    src="https://picsum.photos/300/300"
  />
  <p>
    This is the Home Page
  </p>
</div>
`;
